src/Integration.Vsix/SonarLintIntegrationPackage.cs

```diff
@@ -80,8 +79,6 @@ protected override void Initialize()
 
             IServiceProvider serviceProvider = this;
 
-            this.deprecatedSonarRuleSetManager = this.GetMefService<IDeprecatedSonarRuleSetManager>();
```

Comment:
This line caused the _DeprecatedSonarRulesetManager_ to be created. If the constructor threw then it would crash VS.
The _SingleOrDefault_ call in _GetMefService_ would be on the call stack at the point of the crash. It could be that the unexplained crashes that have been reported are not to do with _SingleOrDefault_ finding multiple items, but errors in the constructors of objects being created via MEF.

I'll create a ticket for hardening the constructors and the package initialisation.

Reply:
Branch needs to be rebased on master to avoid `SingleOrDefault` issues.

Reply:
@Evangelink  I don't think using _FirstOrDefault_ would fix this issue: the problem isn't that we are getting multiple objects returned by MEF, it's that object being created by MEF is throwing an exception which isn't caught.